Pushing large headers results in "AssertionError: Frame too large!"

I've been trying to make CONTINUATION on PUSH_PROMISE work in Firefox. Writing the test for it, I added an endpoint to our test server that does the following:

```javascript
else if (u.pathname === "/continuedheaders") {
  var headers = {'x-pushed-request': 'true',
                 'content-type': 'text/plain',
                 'content-length': '2',
                 'X-Connection-Http2': 'yes'};
  var hdrtxt = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
  for (var i = 0; i < 265; i++) {
    headers['X-Test-Header-' + i] = hdrtxt;
  }
  push = res.push({
    hostname: 'localhost' + serverPort,
    port: serverPort,
    path: '/continuedheaders/push',
    method: 'GET',
    headers: headers
  });
  push.end("ok");
}

res.setHeader('Content-Type', 'text/html');
if (req.httpVersionMajor != 2) {
  res.setHeader('Connection', 'close');
}
res.writeHead(200);
res.end(content);
```

and results in the following error when that code path is hit:

```
0:01.76 LOG: MainThread INFO Node moz-http2 server shutting down ...
 0:01.76 LOG: MainThread INFO Node server moz-http2 already dead 8
 0:01.76 LOG: MainThread INFO Process stderr
 0:01.76 LOG: MainThread INFO
 0:01.76 LOG: MainThread INFO assert.js:93
 0:01.76 LOG: MainThread INFO   throw new assert.AssertionError({
 0:01.76 LOG: MainThread INFO         ^
 0:01.76 LOG: MainThread INFO AssertionError: Frame too large!
 0:01.76 LOG: MainThread INFO     at Serializer._transform (/Users/hurley/src/mozilla/gecko/testing/xpcshell/node-http2/lib/protocol/framer.js:46:3)
 0:01.76 LOG: MainThread INFO     at Serializer.Transform._read (_stream_transform.js:179:10)
 0:01.76 LOG: MainThread INFO     at Serializer.Transform._write (_stream_transform.js:167:12)
 0:01.76 LOG: MainThread INFO     at doWrite (_stream_writable.js:226:10)
 0:01.76 LOG: MainThread INFO     at writeOrBuffer (_stream_writable.js:216:5)
 0:01.76 LOG: MainThread INFO     at Serializer.Writable.write (_stream_writable.js:183:11)
 0:01.76 LOG: MainThread INFO     at write (_stream_readable.js:602:24)
 0:01.76 LOG: MainThread INFO     at flow (_stream_readable.js:611:7)
 0:01.76 LOG: MainThread INFO     at Compressor.pipeOnReadable (_stream_readable.js:643:5)
 0:01.76 LOG: MainThread INFO     at Compressor.emit (events.js:92:17)
 0:01.76 LOG: MainThread INFO     at emitReadable_ (_stream_readable.js:427:10)
 0:01.76 LOG: MainThread INFO     at emitReadable (_stream_readable.js:423:5)
 0:01.76 LOG: MainThread INFO     at readableAddChunk (_stream_readable.js:166:9)
```

This does *not* appear to happen with sending large headers in response to a regular pulled request (in other words, HEADERS + CONTINUATION seems to work fine), nor does this happen with a small-ish PUSH_PROMISE (one that does not require a CONTINUATION), so there's something going wonky in our handling of the PUSH_PROMISE + CONTINUATION path.

--- a/lib/protocol/compressor.js
+++ b/lib/protocol/compressor.js
@@ -1163,7 +1163,11 @@ Compressor.prototype._transform = function _transform(frame, encoding, done) {
   if (frame.type === 'HEADERS' || frame.type === 'PUSH_PROMISE') {
     var buffer = this.compress(frame.headers);
 
-    var chunks = cut(buffer, MAX_HTTP_PAYLOAD_SIZE);
+    // This will result in CONTINUATIONs from a PUSH_PROMISE being 4 bytes shorter than they could
+    // be, but that's not the end of the world, and it prevents us from going over MAX_HTTP_PAYLOAD_SIZE
+    // on the initial PUSH_PROMISE frame.
+    var adjustment = frame.type === 'PUSH_PROMISE' ? 4 : 0;
+    var chunks = cut(buffer, MAX_HTTP_PAYLOAD_SIZE - adjustment);
 
     for (var i = 0; i < chunks.length; i++) {
       var chunkFrame;

